conans/test/functional/toolchains/microsoft/test_msbuild.py

```diff
@@ -607,21 +607,3 @@ def test_toolchain_win_multi(self):
             else:
                 self.assertNotIn("hello.dll", client.out)
             self.assertIn("KERNEL32.dll", client.out)
-
-
-def test_msvc_runtime_flag_common_usage():
```

Comment:
This test was removed, because it was a weird usage of the ``msvc_runtime_flag()`` for validation. This would raise as it is internally accessing ``self.settings``.

If validation of the runtime is desired:
- ``if self.info.settings.get_safe("compiler.runtime") == "static": raise ConanInvalid....`` (only works with ``msvc`` compiler)
- If we need backwards compatibility to Conan 1.X, we need a helper that helps abstracting ``Visual Studio`` and ``msvc`` compilers. ConanCenter has stopped creating ``shared=True`` + VS MT builds, shouldn't be an issue, the 2 recipes that implement this ``validate()`` can remove it.